Tools/AP_Periph/batt_info.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  battery info support. This feature has a set of parameters which are
+  controlled by a lua script on the FMU. The battery infomation node
+  is attached to a specific battery and recorded the number of cycles,
+  and total armed time
+ */
+#include "AP_Periph.h"
+
+#if AP_PERIPH_BATTERY_INFO_ENABLED
+
+extern const AP_HAL::HAL &hal;
+
+const AP_Param::GroupInfo BattInfo::var_info[] {
+    // @Param: _NUM_CYCLES
+    // @DisplayName: Number of cycles
+    // @Description: Number of cycles the battery has been through
+    AP_GROUPINFO("_NUM_CYCLES", 1, BattInfo, num_cycles, 0),
```

Comment:
I guess this is a bit more subtle, hense the need for a remote script? Scripting should run fine on the periph node itself. The script could add the params too.

Reply:
on a L431? I doubt we'd have the memory

Reply:
Maybe, I guess its not actually running anything else? Certainly works on G4's.